crates/sui-node/src/admin.rs

```diff
@@ -0,0 +1,48 @@
+// Copyright (c) 2022, Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use axum::{
+    extract::Extension,
+    http::StatusCode,
+    routing::{get, post},
+    Router,
+};
+use std::net::{IpAddr, Ipv4Addr, SocketAddr};
+use telemetry_subscribers::FilterHandle;
+use tracing::info;
+
+const LOGGING_ROUTE: &str = "/logging";
+
+pub fn start_admin_server(port: u16, filter_handle: FilterHandle) {
+    let app = Router::new()
+        .route(LOGGING_ROUTE, get(get_filter))
+        .route(LOGGING_ROUTE, post(set_filter))
+        .layer(Extension(filter_handle));
+
+    let socket_address = SocketAddr::new(IpAddr::V4(Ipv4Addr::LOCALHOST), port);
+    info!("starting admin server on {}", socket_address);
+
+    tokio::spawn(async move {
+        axum::Server::bind(&socket_address)
+            .serve(app.into_make_service())
+            .await
+            .unwrap();
+    });
+}
+
+async fn get_filter(Extension(filter_handle): Extension<FilterHandle>) -> (StatusCode, String) {
+    match filter_handle.get() {
+        Ok(filter) => (StatusCode::OK, filter),
+        Err(err) => (StatusCode::INTERNAL_SERVER_ERROR, err.to_string()),
+    }
+}
+
+async fn set_filter(
+    Extension(filter_handle): Extension<FilterHandle>,
+    new_filter: String,
+) -> (StatusCode, String) {
+    match filter_handle.update(new_filter) {
+        Ok(()) => (StatusCode::OK, "".into()),
```

Comment:
Perversely enough, would it be possible to add something like `info!("Log EnvFilter string changed to {new_filter}`, I have a feeling that future me, equipped with just a console and no fancy tooling, will appreciate having this anchor in logs as a starting point.

Plus it serves as a trace of what happened in the admin interface: I can envision someone exposing this to the exterior, and a kid passing by discovering they can kill the node my hitting that endpoint with the word "trace".

Reply:
> I can envision someone exposing this to the exterior, and a kid passing by discovering they can kill the node my hitting that endpoint with the word "trace".

Thankfully the code prohibits this as you need to be on localhost to hit it